var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -743,6 +752,21 @@ def define_enable(suffix, value=None):
                 ]
             )
 
+        if "+pytrilinos2" in spec:
+            options.append(define_trilinos_enable("PyTrilinos2", True))
+            if "+test" in spec:
+                options.append(define("PyTrilinos2_ENABLE_TESTS", True))
+            binder_path = spec["binder"].prefix.bin
+            binder = binder_path + "/binder"
+            options.append(define("PyTrilinos2_BINDER_EXECUTABLE", binder))
+            llvm_dir = self.spec["llvm"].prefix
```

Comment:
@jmlapre It's probably cleaner to make llvm an optional dependency of Trilinos.

Reply:
Yes, it's a dependency in the required binder package so it should always be available transitively, right?  PyTrilinos2 requires those to be passed to CMake:

https://github.com/trilinos/Trilinos/blob/develop/packages/PyTrilinos2/README.md

Reply:
Yes, it's correct that binder needs LLVM. But I think that it's cleaner to declare it as a dependency for Trilinos as well in this case.

Reply:
I would lean the other way and try to minimize the possible dependencies.  Trilinos never needs LLVM; binder requires it.  I probably should have left a comment in there explaining where LLVM actually comes from.

Reply:
This really seems like something that should be set in Binding, rather than downstream. For example, IWYU is another utility built on top of clang, and in spack it's set up to link resources in the `iwyu.py` package itself:
```python
    @run_after("install")
    def link_resources(self):
        # iwyu needs to find Clang's headers
        # https://github.com/include-what-you-use/include-what-you-use/blob/master/README.md#how-to-install
        mkdir(self.prefix.lib)
        symlink(self.spec["llvm"].prefix.lib.clang, self.prefix.lib.clang)
```

You're sure there's no way to set up binder so that it works correctly without extra flags?

Reply:
I fixed the code so we only touch `spec["binder"]` now.  I'm not seeing a way to cut down on what we have to pass it though as PyTrilinos2 requires `PyTrilinos2_BINDER_EXECUTABLE`, `PyTrilinos2_BINDER_clang_include_dirs`, and `PyTrilinos2_BINDER_LibClang_include_dir` to be set...